### PR TITLE
projects.get support path encode && fixed tests

### PR DESCRIPTION
**Pull request by atian25**

### CODE REVIEW

File: lib/resources/project.js [hunk 1]
```diff
@@ -33,8 +33,9 @@ util.inherits(Project, restful.RESTFulResource);
  *  - {String} path
  * @param {Function(err, project)} callback
  */
-Project.prototype.getByPath = function (params, callback) {
```
Comment (fengmk2) [inline lib/resources/project.js]: should keep this for compatibility

File: lib/resources/project.js [hunk 2]
```diff
@@ -33,8 +33,9 @@ util.inherits(Project, restful.RESTFulResource);
...
-  this.client.request('get', '/project', params, callback);
+Project.prototype.get = function (params, callback) {
```
Comment (fengmk2) [inline lib/resources/project.js]: I think you should change it from https://github.com/node-modules/restful-client#restfulresource.

`get()` method will auto create by `restful-client`.

Comment (atian25) [inline lib/resources/project.js]: had notify this, but don't know how to wrap origin `get()`, any example?

File: lib/resources/project.js (hunk 1 above)
Comment (atian25) [inline lib/resources/project.js]: did it work long time ago? I see the test case was comment out.
should I just left it or use `get()` inside?

File: lib/resources/project.js (hunk 2 above)
Comment (fengmk2) [inline lib/resources/project.js]: `encodeURIComponent(val)` on this https://github.com/node-modules/restful-client/blob/master/lib/client.js#L121

```js
pathname = pathname.replace(key, encodeURIComponent(val));
```

**Pull request closed without merging**